app/lib/supabase.tsx

```diff
@@ -2,16 +2,32 @@ import React from "react";
 import { createClient, SupabaseClient } from "@supabase/supabase-js";
 import { Auth } from "@supabase/ui";
 
-const SupabaseClientContext = React.createContext<SupabaseClient | null>(null);
+let clientSingleton: SupabaseClient;
```

Comment:
I've done the supabase client this way because this is the same way `next-urql' handles their client on the server so it just felt correct to have the same pattern as they both are intertwined so much